src/analysis.cpp

```diff
@@ -1311,40 +1311,60 @@ void ScatteringFunction::_sample() {
     for (int id : ids) { // loop over molecule names
         auto groups = spc.findMolecules(id);
         for (auto &g : groups) // loop over groups
-            if (usecom && !g.atomic)
+            if (use_com && !g.atomic)
                 p.push_back(g.cm);
             else
                 for (auto &i : g) // loop over particle index in group
                     p.push_back(i.pos);
     }
+
+    // zero-padded suffix to use with `save_after_sample`
+    std::string suffix = std::to_string(cnt);
+    suffix = std::string(7 - suffix.length(), '0') + suffix;
+
     switch (scheme) {
     case DEBYE:
         debye->sample(p, spc.geo.getVolume());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, debye->getIntensity());
         break;
-    case EXPLICIT:
-        explicit_average->sample(p, spc.geo.getLength().x());
+    case EXPLICIT_PBC:
+        explicit_average_pbc->sample(p, spc.geo.getLength().x());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, explicit_average_pbc->getSampling());
+        break;
+    case EXPLICIT_IPBC:
+        explicit_average_ipbc->sample(p, spc.geo.getLength().x());
+        if (save_after_sample)
+            IO::write(filename + "." + suffix, explicit_average_ipbc->getSampling());
         break;
     }
 }
 void ScatteringFunction::_to_json(json &j) const {
-    j = {{"molecules", names}, {"com", usecom}};
+    j = {{"molecules", names}, {"com", use_com}};
     switch (scheme) {
     case DEBYE:
         j["scheme"] = "debye";
-        j["qmin"] = debye->qmin;
-        j["qmax"] = debye->qmax;
+        std::tie(j["qmin"], j["qmax"], std::ignore) = debye->getQMeshParameters();
         break;
-    case EXPLICIT:
+    case EXPLICIT_PBC:
         j["scheme"] = "explicit";
-        j["pmax"] = explicit_average->pmax;
+        j["pmax"] = explicit_average_pbc->getQMultiplier();
+        j["ipbc"] = false;
+        break;
+    case EXPLICIT_IPBC:
+        j["scheme"] = "explicit";
+        j["pmax"] = explicit_average_ipbc->getQMultiplier();
+        j["ipbc"] = true;
         break;
     }
 }
 
 ScatteringFunction::ScatteringFunction(const json &j, Space &spc) try : spc(spc) {
     from_json(j);
     name = "scatter";
-    usecom = j.value("com", true);
+    use_com = j.value("com", true);
+    save_after_sample = j.value("stepsave", false); // save to disk for each sample
```

Comment:
And to `_docs/analysis.md`

Reply:
How come that this could come like this from the pull request #241? :-D Done. Documentation shall also contain `ipbc`. As `ipbc` has no meaning for `debye` scheme, shall we keep it as an independent attribute, or shall we have three different schemes?

Reply:
yes, debye is quite different and also requres dq, qmin, qmx while these are automatic w. explicit and ipbc that om the other hand require qmax. I would keep ipbc as an option in the explicit scheme and split debye to a separate analysis.